### PR TITLE
Comment: add description for some code which hard to understand

### PR DESCRIPTION
**Pull request by HuKeping**
I didn't figure out what it means the first time when  I see the `out.Stream.Write(nil)`, so I think a comment might be help.

Signed-off-by: Hu Keping hukeping@huawei.com

### CODE REVIEW

File: api/server/daemon.go [hunk 1]
```diff
@@ -111,8 +111,12 @@ func (s *Server) getEvents(ctx context.Context, w http.ResponseWriter, r *http.R
 	d := s.daemon
 	es := d.EventsService
 	w.Header().Set("Content-Type", "application/json")
+
 	outStream := ioutils.NewWriteFlusher(w)
-	outStream.Write(nil) // make sure response is sent immediately
+	// Write an empty chunk of data.
+	// This is to ensure that the HTTP Response is sent immediately,
```
Comment (duglin) [inline api/server/daemon.go]: s/Response/Status Code/   I think is a bit clearer since the full response is streamed after this.

**Comment by HuKeping** [comment]
updated @duglin

**Comment by duglin** [comment]
LGTM

File: api/server/daemon.go [hunk 2]
```diff
@@ -111,8 +111,12 @@ func (s *Server) getEvents(ctx context.Context, w http.ResponseWriter, r *http.R
 	d := s.daemon
 	es := d.EventsService
 	w.Header().Set("Content-Type", "application/json")
+
 	outStream := ioutils.NewWriteFlusher(w)
-	outStream.Write(nil) // make sure response is sent immediately
+	// Write an empty chunk of data.
+	// This is to ensure that the HTTP status code is sent immediately,
+	// even if the container has not yet produced any data.
```
Comment (cpuguy83) [inline api/server/daemon.go]: This last line doesn't make any sense as this is not related to a container.

Comment (tiborvass) [inline api/server/daemon.go]: @HuKeping I agree with @cpuguy83, Events is not specific to containers.

**Comment by HuKeping** [comment]
@cpuguy83 @tiborvass updated.

**Comment by cpuguy83** [comment]
LGTM
